Tests/Services/NfieldSurveyResourcesServiceTests.cs

```diff
@@ -69,21 +69,24 @@ public void TestQueryAsync_ServerReturnsQuery_ReturnsListWithSurveyResources()
             var target = new NfieldSurveyResourcesService();
             target.InitializeNfieldConnection(mockedNfieldConnection.Object);
 
-            var actualSurveyResources = target.QueryAsync().Result;
+            var actualSurveyResources = await target.QueryAsync();
 
             Assert.Equal(5, actualSurveyResources.Count());
 
             for (var i = 0; i < 5; i++)
             {
-                Assert.Equal(expectedSurveyResources[i].SurveyId, actualSurveyResources.ToArray()[i].SurveyId);
-                Assert.Equal(expectedSurveyResources[i].SurveyName, actualSurveyResources.ToArray()[i].SurveyName);
-                Assert.Equal(expectedSurveyResources[i].Owner, actualSurveyResources.ToArray()[i].Owner);
-                Assert.Equal(expectedSurveyResources[i].Size, actualSurveyResources.ToArray()[i].Size);
-                Assert.Equal(expectedSurveyResources[i].CreationDate, actualSurveyResources.ToArray()[i].CreationDate);
-                Assert.Equal(expectedSurveyResources[i].LastDataDownloadDate, actualSurveyResources.ToArray()[i].LastDataDownloadDate);
-                Assert.Equal(expectedSurveyResources[i].LastDataCollectionDate, actualSurveyResources.ToArray()[i].LastDataCollectionDate);
-                Assert.Equal(expectedSurveyResources[i].State, actualSurveyResources.ToArray()[i].State);
-                Assert.Equal(expectedSurveyResources[i].Channel, actualSurveyResources.ToArray()[i].Channel);
+                var actualSurveyResource = actualSurveyResources.ToArray()[i];
```

Comment:
Maybe you can do actualSurveyResources.ElementAt(i). Now you are converting it to an array for each iteration of the loop